core/domain/stats_jobs_continuous.py

```diff
@@ -526,41 +526,118 @@ def entity_classes_to_map_over(cls):
     # recomputing results from scratch each time.
     @staticmethod
     def map(item):
-        def _encode(item):
-            """Encodes a Python value as UTF-8."""
-            if isinstance(item, basestring):
-                return item.encode('utf-8')
-            elif isinstance(item, int) or isinstance(item, float):
-                return item
-            elif isinstance(item, list):
-                return [_encode(subitem) for subitem in item]
-            elif isinstance(item, dict):
-                return {
-                    key: _encode(value) for key, value in item.iteritems()
-                }
-            else:
-                raise Exception('Cannot encode item: %s' % item)
-
         if InteractionAnswerSummariesMRJobManager._entity_created_before_job_queued( # pylint: disable=line-too-long
                 item):
             # Output answers submitted to the exploration for this exp version.
-            versioned_key = u'%s:%s:%s:%s' % (
-                item.exploration_id, item.exploration_version,
-                item.interaction_id, item.state_name)
-            yield (versioned_key.encode('utf-8'), item.id)
+            versioned_key = u'%s:%s:%s' % (
+                item.exploration_id, item.exploration_version, item.state_name)
+            yield (versioned_key.encode('utf-8'), {
+                'item_id': item.id,
```

Comment:
"item" is really too vague, and I keep forgetting what item we are talking about. Can we call it state_answers_model_id perhaps?